api/src/main/java/org/openmrs/module/reporting/dataset/definition/evaluator/PatientDataSetEvaluator.java

```diff
@@ -90,6 +90,7 @@ public DataSet evaluate(DataSetDefinition dataSetDefinition, EvaluationContext c
 			StopWatch sw = new StopWatch();
 			sw.start();
 
+			ec.addContextValue("columnDefinitions", cd);
```

Comment:
Can you describe why this is needed?  Is there a ticket for this change?  What problem are you trying to solve?

Reply:
This was needed within the sequence of an issue identified on [patient-grid](https://github.com/icrc/openmrs-module-patientgrid/). The patient-grid uses module-reporting to retrieve data for populating the grid. An issue arises when the same concept is reused within a single form. patient-grid lacks sufficient data to correlate observations with their respective columns.

On patient-grid, module-reporting DataSetDefinitionServiceImpl is invoked [here](https://github.com/icrc/openmrs-module-patientgrid/blob/173fde6672e86ceec2380ee25af2155f82bf7958/api/src/main/java/org/openmrs/module/patientgrid/api/impl/PatientGridServiceImpl.java#L100C3-L100C3) and, on the 1st arg, column data is passed. Foreach column, reporting-module invokes patient-grid's [ObsForLatestEncounterPatientDataEvaluator.evaluate](https://github.com/icrc/openmrs-module-patientgrid/blob/173fde6672e86ceec2380ee25af2155f82bf7958/api/src/main/java/org/openmrs/module/patientgrid/evaluator/ObsForLatestEncounterPatientDataEvaluator.java#L26). However, the challenge here is that there is no provided data that allows us to identify the specific column being processed. Currently, fetching observations relies solely on encounter and concept, which can lead to multiple matching observations, differing only in their [formNamespaceAndPath](https://github.com/openmrs/openmrs-core/blob/4c1344f5496fdb420c26d9314ac3bf280247a790/api/src/main/java/org/openmrs/BaseFormRecordableOpenmrsData.java#L33C40-L33C40).

The inclusion of column data within the [ObsForLatestEncounterPatientDataEvaluator.evaluate](https://github.com/icrc/openmrs-module-patientgrid/blob/173fde6672e86ceec2380ee25af2155f82bf7958/api/src/main/java/org/openmrs/module/patientgrid/evaluator/ObsForLatestEncounterPatientDataEvaluator.java#L26) will allow us to correlate the obs formNamespaceAndPath with the right column.

Reply:
Thanks @icrc-psousa .  Could you configure things to pass the necessary information as a configuration parameter from your data set definition down to your data definition.  Can you show me where these data set and data definitions are constructed and configured?

Reply:
Data set definition is constructed [here](https://github.com/icrc/openmrs-module-patientgrid/blob/cfc58f8fad037dec3574415e7f406a9bb8d726d1/api/src/main/java/org/openmrs/module/patientgrid/PatientGridUtils.java#L75). I missed the fact that it's populated on patient-grid side.
We can include the `formNamespaceAndPath` and other required data, if needed, on patient-grid side here on this [data definition](https://github.com/icrc/openmrs-module-patientgrid/blob/cfc58f8fad037dec3574415e7f406a9bb8d726d1/api/src/main/java/org/openmrs/module/patientgrid/PatientGridUtils.java#L149) as you've suggested. Thanks for your input @mseaton!

This PR on module-reporting won't be needed.